### ISSUE TITLE
ValueError

### ISSUE THREAD

**Issue by ashokpatowary** [mortazavilab/swan_vis#5]
Hi @fairliereese 

I have generated the gtf files from talon database. I am encountering the following errors. Can you please have a look in it.

```
Novelty types between datasets conflict. Strongly consider using input from the same data source to reconcile these. Conflicting isoforms will be labelled "Ambiguous".
Traceback (most recent call last):
  File "swan.p", line 21, in <module>
    count_cols='vRG_370')
  File "/u/home/a/ashokpat/.local/lib/python3.7/site-packages/swan_vis/swangraph.py", line 187, in add_dataset
    self.update_ids()
  File "/u/home/a/ashokpat/.local/lib/python3.7/site-packages/swan_vis/graph.py", line 120, in update_ids
    self.dfs_to_dicts()
  File "/u/home/a/ashokpat/.local/lib/python3.7/site-packages/swan_vis/graph.py", line 231, in dfs_to_dicts
    self.t_df = self.t_df.to_dict('index')
  File "/u/home/a/ashokpat/.local/lib/python3.7/site-packages/pandas/core/frame.py", line 1391, in to_dict
    raise ValueError("DataFrame index must be unique for orient='index'.")
ValueError: DataFrame index must be unique for orient='index'.
```

**Comment by fairliereese** [comment]
Hi there. Are you using GTFs from different TALON databases?

**Comment by ashokpatowary** [comment]
I am using GTF generated from same database. I had the same error when I tried to run in from the talon database.

**Comment by fairliereese** [comment]
Ugh, I'm sorry. This problem happens when you have two transcripts with different exon connectivity that have the same name. Did you already add an annotation to the SwanGraph?

**Comment by ashokpatowary** [comment]
Yah, I add the genecode v33 SwanGraph.

**Comment by fairliereese** [comment]
Could you send me your GTFs so I can check them out?

**Comment by ashokpatowary** [comment]
Yah Sure.

**Comment by fairliereese** [comment]
I'll take a look into it first thing tomorrow.

**Comment by fairliereese** [comment]
Hi Ashok. After looking into this, it seems like you are actually using the GENCODE v33 lifted over annotation, which I think we've discussed in the past, correct? 

Unfortunately I have discovered that the lifted over version of the annotation has the exons listed in the GTF in the opposite order than Swan expects for minus strand transcripts, even though the normal version of the annotation does not. Currently, Swan relies on this ordering. 

In your case, I see two options. The first is to rerun TALON using the normal GENCODE v33 annotation instead of the liftover version so that you can use the normal GENCODE v33 annotation in Swan. The second is that I will take a stab at removing Swan's expectation that exons be listed in a certain order next week, if you'd prefer to wait. I'd advise rerunning TALON personally though because I can't guarantee that things will work out quickly on my end.

Here, I'm just demonstrating the differences in the GTF ordering.
In the TALON GTF:
```
chr6	HAVANA	transcript	143,815,949	143,832,857	-
chr6	HAVANA	exon	143,832,548	143,832,857	-
chr6	HAVANA	exon	143,828,374	143,828,561	-
chr6	HAVANA	exon	143,825,050	143,825,389	-
chr6	HAVANA	exon	143,823,492	143,823,702	-
chr6	HAVANA	exon	143,823,069	143,823,259	-
chr6	HAVANA	exon	143,818,526	143,818,634	-
chr6	HAVANA	exon	143,815,949	143,816,984	-
```

In the GENCODE lifted over reference:
```
chr6	HAVANA	transcript	143,815,949	143,832,857	-
chr6	HAVANA	exon	143,815,949	143,816,984	-
chr6	HAVANA	exon	143,818,526	143,818,634	-
chr6	HAVANA	exon	143,823,069	143,823,259	-
chr6	HAVANA	exon	143,823,492	143,823,702	-
chr6	HAVANA	exon	143,825,050	143,825,389	-
chr6	HAVANA	exon	143,828,374	143,828,561	-
chr6	HAVANA	exon	143,832,548	143,832,857	-
```

**Comment by ashokpatowary** [comment]
@fairliereese 

Thanks for the nice explanation. All, my analysis is on GRCh37 so I have no other choice but to use the lift over version of Gencode. But, I will try to use the GRCh38 one time and see how its looks. Anyways thanks again.

**Comment by fairliereese** [comment]
This has now been patched as of version 1.0.2.
